src/bsoncxx/json.hpp

```diff
@@ -34,18 +34,39 @@ BSONCXX_INLINE_NAMESPACE_BEGIN
 extern const bool k_silence_doxygen;
 
 ///
-/// Converts a BSON document to a JSON string.
+/// An enumeration of the types of Extended JSON that the to_extended_json function accepts
+///
+/// The values are:
+///   k_canonical: to produce Canonical Extended JSON
+///   k_relaxed: to produce Relaxed Extended JSON
+enum class ExtendedJsonMode { k_canonical, k_relaxed };
+
+///
+/// Converts a BSON document to a JSON string, in legacy extended format.
 ///
 /// @param view
 ///   A valid BSON document.
 ///
 /// @throws bsoncxx::exception with error details if the conversion failed.
 ///
-/// @returns A JSON string.
+/// @returns A legacy extended JSON string.
```

Comment:
Should their also be an enumerator value that causes this function to be called? If the old format was 'extended', as the comment suggests, then surely it should be possible to obtain that behavior via the new `to_extended_json` API as well.

And, if you did that, you could mark this function as deprecated.

Reply:
For future applications, "legacy" extended shouldn't really be used at all, so I am happy to mark this as "deprecated", but wouldn't want to add it to the enum regardless.

Reply:
By doing this, you are forcing people to change both what function they use, and what format they get, at the same time. Giving them a nicer transitional path is better. I can move to the non-deprecated function but retain the format I want/expect. Then later I can upgrade to the new format when my application or other consumers are ready to deal. Why force both changes at once?